### ISSUE TITLE
when create/register a new user, user's avatar filename is not right in file system

### ISSUE THREAD

**Issue by qianlongzt** [go-gitea/gitea#3515]
- Gitea version (or commit ref): 1.4.0-rc1
- Git version: 2.16.1
- Operating system: Ubuntu 16.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:
register user
```
2018/02/15 15:22:56 [I] New random avatar created: 31
2018/02/15 15:22:57 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:22:57 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:22:57 [D] Template: status/404
2018/02/15 15:22:57 [I] New random avatar created: 31
2018/02/15 15:22:57 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:22:57 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:06 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:06 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:06 [D] Template: user/dashboard/dashboard
2018/02/15 15:23:06 [I] New random avatar created: 31
2018/02/15 15:23:06 [I] New random avatar created: 31
2018/02/15 15:23:06 [I] New random avatar created: 31
2018/02/15 15:23:07 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:07 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:07 [D] Template: status/404
2018/02/15 15:23:07 [I] New random avatar created: 31
2018/02/15 15:23:07 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:07 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:07 [D] Template: status/404
2018/02/15 15:23:07 [I] New random avatar created: 31
2018/02/15 15:23:07 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:07 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:16 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:16 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:16 [D] Template: user/profile
2018/02/15 15:23:16 [I] New random avatar created: 31
2018/02/15 15:23:16 [I] New random avatar created: 31
2018/02/15 15:23:16 [I] New random avatar created: 31
2018/02/15 15:23:16 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:16 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:16 [D] Template: status/404
2018/02/15 15:23:16 [I] New random avatar created: 31
2018/02/15 15:23:16 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:16 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:16 [D] Template: status/404
2018/02/15 15:23:16 [I] New random avatar created: 31
2018/02/15 15:23:20 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:20 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:20 [D] Template: status/404
2018/02/15 15:23:20 [I] New random avatar created: 31
2018/02/15 15:23:20 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:23:20 [D] CSRF Token: O9RMhHyhG8gVGRDQZDSzYo_Epd86MTUxODY3OTM3NjczODEzNDE0Mw==
2018/02/15 15:23:20 [D] Template: status/404
2018/02/15 15:23:20 [I] New random avatar created: 31
```

create user by admin 
```
2018/02/15 15:46:18 [D] Template: admin/user/new
2018/02/15 15:46:30 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:46:30 [D] CSRF Token: 1XiAkJhaWEAlsZNb_Di3MZUUVXQ6MTUxODY3OTY2MTMzMjA0Mzk4OQ==
2018/02/15 15:46:30 [T] Account created by admin (qianlongzt): bbb
2018/02/15 15:46:30 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:46:30 [D] CSRF Token: 1XiAkJhaWEAlsZNb_Di3MZUUVXQ6MTUxODY3OTY2MTMzMjA0Mzk4OQ==
2018/02/15 15:46:30 [D] Template: admin/user/edit
2018/02/15 15:46:35 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:46:35 [D] CSRF Token: 1XiAkJhaWEAlsZNb_Di3MZUUVXQ6MTUxODY3OTY2MTMzMjA0Mzk4OQ==
2018/02/15 15:46:35 [D] Template: user/profile
2018/02/15 15:46:35 [I] New random avatar created: 32
2018/02/15 15:46:35 [I] New random avatar created: 32
2018/02/15 15:46:35 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:46:35 [D] CSRF Token: 1XiAkJhaWEAlsZNb_Di3MZUUVXQ6MTUxODY3OTY2MTMzMjA0Mzk4OQ==
2018/02/15 15:46:35 [D] Template: status/404
2018/02/15 15:46:42 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:46:42 [D] CSRF Token: 1XiAkJhaWEAlsZNb_Di3MZUUVXQ6MTUxODY3OTY2MTMzMjA0Mzk4OQ==
2018/02/15 15:46:42 [D] Template: status/404
2018/02/15 15:46:46 [D] Session ID: 6846fe07f43568b5
2018/02/15 15:46:46 [D] CSRF Token: 1XiAkJhaWEAlsZNb_Di3MZUUVXQ6MTUxODY3OTY2MTMzMjA0Mzk4OQ==
2018/02/15 15:50:43 [D] Session ID: dc6264f689045100
2018/02/15 15:50:43 [D] CSRF Token: jHvViyIg_9z3DtpwX1FUyTWBbCI6MTUxODY4MTA0MzMzOTA4MjE0Mw==
2018/02/15 15:50:43 [D] Template: home

```
## Description
1. create or register a new user, user's avatar filename is UserId in file system, but avatar is a md5 hash in database. Screenshots 1,2
1. refresh user's home page(such as `https://example.com/bbb`), system will generate a new avatar and save it in a filed named UserId.  Screenshots 3,4
1. but in https://try.gitea.io all is right.

## Screenshots

![image](https://user-images.githubusercontent.com/18493471/36246531-3fc272ce-126b-11e8-9ce9-622240f5772e.png)
![image](https://user-images.githubusercontent.com/18493471/36246566-65e60128-126b-11e8-95bb-397ddb31eaa4.png)

![image](https://user-images.githubusercontent.com/18493471/36246305-1e6091fc-126a-11e8-82c7-6fe2049f171a.png)
![image](https://user-images.githubusercontent.com/18493471/36246328-335b9b1a-126a-11e8-8ce4-4b2616ab2433.png)

**Comment by qianlongzt** [comment]
use custom avatar, everything is ok

**Comment by cody-vibe** [comment]
This might be a duplicate of #3464. A patch was merged in both `master` and `release/v1.4`. Can you please compile from source and check if it's fixed?

**Comment by qianlongzt** [comment]
it's fixed in release/v1.4
